### PR TITLE
OKTA-628971: Refactor redirects script

### PR DESCRIPTION
**Pull request by paulwallace-okta**

### CODE REVIEW

File: scripts/redirect_url_utils.py [hunk 1]
```diff
@@ -83,18 +103,27 @@ def save_map():
...
-      help='apply_map: load map and create redirects. save_map: create file to url map',
-      nargs='?',
-      default='apply_map')
-  args = parser.parse_args()
-
-  if args.method == 'save_map':
+def run(file=None, save=None):
+  if file:
+    print("Adding new redirects...")
+    update_map(file)
```
Comment (emikhasyak-okta) [inline scripts/redirect_url_utils.py:109]: The next time we call `save_map()`, will it override what was added using `update_map` method?

File: scripts/redirect_url_utils.py [hunk 2]
```diff
@@ -25,11 +25,31 @@
...
     <body></body>
-</html>'''
+</html>
+'''
 
 def apply_map():
+  _add_redirects_from_file(FILE_TO_URL)
+
+def update_map(file):
+  """Update the main redirects file with the new redirect pairs in 'file'."""
```
Comment (emikhasyak-okta) [inline scripts/redirect_url_utils.py:35]: I thought your goal was to modify how we create redirect files from the stored map, that is where you see 8k files modified right?

File: scripts/redirect_url_utils.py [hunk 3]
```diff
@@ -25,11 +25,31 @@
...
+  if not os.path.isfile(file):
+    print("'%s' is not a file. Try again." % file)
+    return False
+  _add_redirects_from_file(file)
```
Comment (emikhasyak-okta) [inline scripts/redirect_url_utils.py:39]: ah, that is where new redirects are added, then we update the map.

so the next call to `save_map` should pick newly created redirects and the file with mapping will remain the same.

File: scripts/redirect_url_utils.py [hunk 4]
```diff
@@ -25,11 +25,31 @@
...
+  with open(FILE_TO_URL, 'r', encoding='utf-8') as f:
+    redirects = f.readlines()
+  with open(file, 'r', encoding='utf-8') as f:
+    new_redirects = f.readlines()
+  for new_re in new_redirects:
+    redirects.append(new_re)
+  with open(FILE_TO_URL, 'w', encoding='utf-8') as w:
+    for r in redirects:
+      w.write(r)
```
Comment (emikhasyak-okta) [inline scripts/redirect_url_utils.py:48]: nit:
Lines 40-48 - can we just read new redirects and append to an existing file? 
```
with open(file, 'r', encoding='utf-8') as f:
  new_redirects = f.readlines()
with open(FILE_TO_URL, 'a', encoding='utf-8') as w:
    for r in new_redirects:
        w.write(r)
```

File: scripts/redirect_url_utils.py (hunk 2 above)
Comment (emikhasyak-okta) [reply scripts/redirect_url_utils.py:35]: ignore this one

File: scripts/redirect_url_utils.py (hunk 3 above)
Comment (emikhasyak-okta) [reply scripts/redirect_url_utils.py:39]: ignore this one

File: scripts/redirect_url_utils.py (hunk 1 above)
Comment (emikhasyak-okta) [reply scripts/redirect_url_utils.py:109]: ignore this one

**Review by emikhasyak-okta — APPROVED** [review]
Nice update!
I left a couple of comments while was figuring out the algorithm, pls ignore all except one regarding algorithm improvement 😄 
🚢

File: scripts/redirect_url_utils.py (hunk 4 above)
Comment (paulwallace-okta) [reply scripts/redirect_url_utils.py:48]: Yep, definitely better, thanks much

**Pull request merged**